cloudinit/config/cc_refresh_rmc_and_interface.py

```diff
@@ -0,0 +1,99 @@
+# (c) Copyright IBM Corp. 2020 All Rights Reserved
+#
+# Author: Aman Kumar Sinha <amansi26@in.ibm.com>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+"""
+Refresh IPv6 interface and RMC
+------------------------------
+**Summary:** Ensure Network Manager is not managing IPv6 interface
+
+This module is IBM PowerVM Hypervisor specific
+
+Reliable Scalable Cluster Technology (RSCT) is a set of software components
+that together provide a comprehensive clustering environment(RAS features)
+for IBM PowerVM based virtual machines. RSCT includes the Resource
+Monitoring and Control (RMC) subsystem. RMC is a generalized framework used
+for managing, monitoring, and manipulating resources. RMC runs as a daemon
+process on individual machines and needs creation of unique node id and
+restarts during VM boot.
+More details refer
+https://www.ibm.com/support/knowledgecenter/en/SGVKBA_3.2/admin/bl503_ovrv.htm
+
+This module handles
+- Refreshing RMC
+- Disabling Network Manager from handling IPv6 interface, as IPv6 interface
+  is used for communication between RMC daemon and PowerVM hypervisor.
+
+**Internal name:** per  always
+
+**Supported distros:** RHEL
+
+"""
+
+from cloudinit.settings import PER_ALWAYS
+from cloudinit import util
+from cloudinit import netinfo
+from cloudinit import subp
+
+frequency = PER_ALWAYS
+
+RMCCTRL = '/opt/rsct/bin/rmcctrl'
+SED = '/usr/bin/sed'
+
+
+def handle(name, _cfg, _cloud, log, _args):
+    info = netinfo.netdev_info()
+    # Setting NM_CONTROLLED=no for IPv6 interface
+    # making it down and up
+    info_len = len(info.keys())
+
+    log.debug('Making the ipv6 up')
+    for info_count in range(info_len):
+        try:
+            if 'lo' in list(info.keys())[info_count]:
+                log.debug('Skipping localhost interface')
+                continue
+            interface = list(info.keys())[info_count]
+            if not info[str(list(info.keys())[info_count])]['ipv4']:
+                disable_NM_ipv6(interface)
```

Comment:
According to the description on the header, you expect the VM to have a single IPv6 interface and that's intended to be used to talk to the RMC. Is there a corner case that the VM needs to have more than one IPv6 interface, for example? Can you pinpoint which interface the VM uses to talk to RMC?

Reply:
@otubo , the hypervisor specifically uses ipv6 link local address / interface for RMC communication from the VM to the hypervisor . The expectation here is that when the VM is deployed, the VM is deployed with a ipv6 link local address that is used for this communication. In a case the VM has more than one link local ipv6 address, the hypervisor has the option to use any of these for communication. 
https://www.ibm.com/support/knowledgecenter/POWER8/p8eig/p8eig_update_rmc.htm

Resource Monitoring Control (RMC) connections between PowerVM® NovaLink and each logical partition are routed through a dedicated internal virtual network. For easier configuration of the internal virtual network, assign an IPv6 link-local address to the RMC virtual network interfaces. To assign IPv6 link-local addresses to the RMC virtual network interfaces on AIX® and Linux logical partitions, the Reliable Scalable Cluster Technology (RSCT) packages must be at version 3.2.1.0 or later.